### PR TITLE
small grammar fixes, reporting fixes

### PR DESCRIPTION
**Pull request by rcv-aws**
### Description
Grammar fixes & reporting fixes
 
### Issues Resolved
- grammar fixes for nested RETURN in UDFs
- grammar fixes for ALTER PARITION
- ABORT can be a keyword
- small reporting fixes
- fixed lookup of identifiers with embedded special chars
- in xref, add list of object names with/without issues
 
### Check List
- [ x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish_internal/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/compass/CompassUtilities.java [hunk 1]
```diff
@@ -4552,11 +4564,11 @@ else if (subContext.startsWith("TABLE ")) {
...
-		for (String t : objTypeIssueMap.keySet()) {
-			Integer iX = objTypeIssueCount.getOrDefault(t,0);
-			Integer i0 = objTypeNoIssueCount.getOrDefault(t,0);
-			//appOutput(thisProc()+"t=["+t+"] iX=["+iX+"] i0=["+i0+"] ");
-		}
+//		for (String t : objTypeIssueMap.keySet()) {
+//			Integer iX = objTypeIssueCount.getOrDefault(t,0);
+//			Integer i0 = objTypeNoIssueCount.getOrDefault(t,0);
+//			appOutput(thisProc()+"t=["+t+"] iX=["+iX+"] i0=["+i0+"] ");
+//		}
```
Comment (laurenz) [inline src/main/java/compass/CompassUtilities.java:4571]: The same applies for this: commented-out code is from the time before there was version control.

**Review by laurenz — CHANGES_REQUESTED** [review]

File: src/main/java/compass/CompassAnalyze.java [hunk 2]
```diff
@@ -3549,14 +3549,16 @@ private void captureTriggerOptions(String type, String trigName, List<TSQLParser
 					if (u.debugging) u.dbgOutput(CompassUtilities.thisProc()+"UDF "+ ctx.getText()+", funcName=["+funcName+"] sudfDataType=["+sudfDataType+"] ", u.debugPtree);
 					captureItem(sudfDataType + sudfDataTypeOrig + " scalar function result type", "", Datatypes, getBaseDataType(sudfDataType), statusDataType, ctx.start.getLine());
 
-					if (ctx.func_body_returns_scalar().RETURN() != null) {
-						captureItem("RETURN"+" scalar, in function", "", ControlFlowReportGroup, "RETURN", u.Supported, ctx.func_body_returns_scalar().RETURN().getSymbol().getLine());
-					}
+//					if (ctx.func_body_returns_scalar().RETURN() != null) {
+						// this is captured as a RETURN statement now
+//						captureItem("RETURN"+" scalar, in function", "", ControlFlowReportGroup, "RETURN", u.Supported, ctx.func_body_returns_scalar().RETURN().getSymbol().getLine());
+//					}
```
[the remark this answers is not in the document]
Comment (rcv-aws) [reply src/main/java/compass/CompassAnalyze.java:3555]: the reason for leaving this in is that it clarifies some of the context in which this happens

File: src/main/java/compass/CompassUtilities.java (hunk 1 above)
Comment (rcv-aws) [reply src/main/java/compass/CompassUtilities.java:4571]: this code is used for debugging. I will add a comment clarifying that.
but more generally, there is a lot of scope for code cleanup and overall rationalization in Babelfish Compass. This is just one case. That would be a sub-project by itself, and currently that does not have priority.

**Review by hong-yuan — APPROVED** [review]

**Comment by laurenz** [comment]
Hm. I think that debugging code should not be commented out, but be conditional on a debugging flag.
But reality is not always perfect, so I'll let it go.

**Review by laurenz — APPROVED** [review]

**Pull request merged**